src/lib/update-lock.ts

```diff
@@ -152,20 +103,41 @@ export function lock<T extends unknown>(
 								return Bluebird.mapSeries(
 									lockFilesOnHost(appId, serviceName),
 									(tmpLockName) => {
-										return Bluebird.try(() => {
-											if (force || lockOverride) {
-												return lockFile.unlockAsync(tmpLockName);
-											}
-										})
-											.then(() => lockFile.lockAsync(tmpLockName))
-											.then(() => {
-												locksTaken[tmpLockName] = true;
+										return (
+											Bluebird.try(() => {
+												if (force || lockOverride) {
+													return lockfile.unlock(tmpLockName);
+												}
 											})
-											.catchReturn(ENOENT, undefined);
+												.then(() => {
+													return lockfile.lock(tmpLockName);
+												})
+												// If lockfile exists, dispose of writeLock and throw a user-friendly error
+												.catch(lockfile.LockfileExistsError, () => {
+													return dispose(release).throw(
+														new UpdatesLockedError(
+															`Lockfile exists for ${JSON.stringify({
+																serviceName,
+																appId,
+															})}`,
+														),
+													);
+												})
+												// Else, dispose of writeLock and throw error as-is
```

Comment:
Do we need this extra `catch`? What happens if you don't add this? Where is the error handled?

Reply:
Before I added it because of the `dispose(release)` call, but after moving the `dispose(release)` call to `.finally` this 2nd catch is not needed anymore.